src/transports/quic.rs

```diff
@@ -0,0 +1,180 @@
+use std::{collections::HashMap, net::SocketAddr, thread::JoinHandle, time::Duration};
+
+use crossbeam::sync::WaitGroup;
+use massa_signature::{PUBLIC_KEY_SIZE_BYTES, SIGNATURE_SIZE_BYTES};
+use mio::{net::UdpSocket, Events, Interest, Poll, Token, Waker};
+use quiche::{accept, connect, ConnectionId};
+
+use crate::{
+    endpoint::Endpoint, error::PeerNetError, network_manager::SharedPeerDB, peer::Peer,
+    peer_id::PeerId,
+};
+
+use super::Transport;
+
+const NEW_CONNECTION_SERVER: Token = Token(0);
+const STOP_LISTENER: Token = Token(10);
+
+pub(crate) struct QuicTransport {
+    pub peer_db: SharedPeerDB,
+    pub out_connection_attempts: WaitGroup,
+    pub listeners: HashMap<SocketAddr, (Waker, JoinHandle<()>)>,
+}
+
+#[derive(Clone)]
+pub struct QuicOutConnectionConfig {
+    pub identity: PeerId,
+    pub local_addr: SocketAddr,
+}
+
+impl QuicTransport {
+    pub fn new(peer_db: SharedPeerDB) -> QuicTransport {
+        QuicTransport {
+            peer_db,
+            out_connection_attempts: WaitGroup::new(),
+            listeners: Default::default(),
+        }
+    }
+}
+
+impl Transport for QuicTransport {
+    type OutConnectionConfig = QuicOutConnectionConfig;
+
+    fn start_listener(&mut self, address: SocketAddr) -> Result<(), PeerNetError> {
+        let mut poll = Poll::new().map_err(|err| PeerNetError::ListenerError(err.to_string()))?;
+        let mut events = Events::with_capacity(128);
```

Comment:
May be add it as constant or config property in the future

Reply:
Added a todo